File: c/605.c
bool canPlaceFlowers(int* f, int l, int n) {
    for (int i = f[0] * 2; i < l && n > 0; i++)
        if (f[i] == 0 && (i == l - 1 || f[i + 1] == 0)) {
            n--;
            i++;
        } else if (f[i] == 1)
            i++;
    return n == 0;
}

File: python3/605.py
class Solution:
    def canPlaceFlowers(self, f: List[int], n: int) -> bool:
        l = len(f)
        i = f[0] * 2
        while i < l and n > 0:
            if f[i] == 0 and (i == l - 1 or f[i + 1] == 0):
                n -= 1
                i += 1
            elif f[i] == 1:
                i += 1
            i += 1
        return n == 0
